### ISSUE TITLE
Feature request: Advanced option to manually edit the docker run command

### ISSUE THREAD

**Issue by alixplayz** [pterodactyl/panel#3867]
### Is there an existing feature request for this?

- [X] I have searched the existing issues before opening this feature request.

### Describe the feature you would like to see.

I would like there to be a section on each server where you can manually go in and edit/add args to the docker run command. This would ofc be an advanced option for advanced users with some disclaimer next to it. 

### Describe the solution you'd like.

Useful if you want to add really niche specific things to the docker container.

**Comment by Kieranwest** [comment]
+1 for this, this would also help resolve #3865

**Comment by matthewpi** [comment]
We don't start containers with a `docker run` command, we use direct API calls.  While this is still possible, I'd rather not directly expose container options to the panel as that is just asking for problems.  The ideal solution is to request individual features/flags to be exposed, rather than just a blanket "let us do whatever we want".  If you need full control over what options a container is started with then we are either missing a feature within Pterodactyl for the specific options, or you are trying to use this software in a way that is not intended.

This is especially dangerous as currently the admin panel has no permissions at all, meaning anyone with an admin API key or root admin access could create privileged containers that run as root and compromise every node connected to the panel with minimal effort.

TL;DR I am fine with implementing specific options and flags, not with giving full control over container options.

See the following issues:
https://github.com/pterodactyl/panel/issues/3547
https://github.com/pterodactyl/panel/issues/3545
https://github.com/pterodactyl/panel/issues/3100
https://github.com/pterodactyl/panel/issues/1703
https://github.com/pterodactyl/panel/issues/1418

**Comment by Kieranwest** [comment]
Would it then be possible to consider exposing the --ip flag within the panel ? As I'd like to be able to use my game servers without needing to share the hosts IP and instead provide the ability to directly assign an IP from my LAN. I did briefly attempt to use other drivers within the config.yml but that just resulted in random IPs that may have already been used elsewhere.

**Comment by matthewpi** [comment]
> Would it then be possible to consider exposing the --ip flag within the panel ? As I'd like to be able to use my game servers without needing to share the hosts IP and instead provide the ability to directly assign an IP from my LAN. I did briefly attempt to use other drivers within the config.yml but that just resulted in random IPs that may have already been used elsewhere.

Let's discuss that in https://github.com/pterodactyl/panel/issues/3865, not here.

**Comment by alixplayz** [comment]
> We don't start containers with a `docker run` command, we use direct API calls. While this is still possible, I'd rather not directly expose container options to the panel as that is just asking for problems. The ideal solution is to request individual features/flags to be exposed, rather than just a blanket "let us do whatever we want". If you need full control over what options a container is started with then we are either missing a feature within Pterodactyl for the specific options, or you are trying to use this software in a way that is not intended.
> 
> This is especially dangerous as currently the admin panel has no permissions at all, meaning anyone with an admin API key or root admin access could create privileged containers that run as root and compromise every node connected to the panel with minimal effort.
> 
> TL;DR I am fine with implementing specific options and flags, not with giving full control over container options.
> 
> See the following issues: #3547 #3545 #3100 #1703 #1418

My issue is that I need to --expose a port, but not -p it. Currently when a port is allocated to the server through the panel it adds that port to --expose and -p. It'd be nice to choose either one.

**Comment by djzoneporro** [comment]
I have the same problem with a custom bridge but I want to assign static ip's to each docker container and in pterodactyl I can't set --ip. 
Is there a way to assign pterodactyl containers a static ip?
